Jellyfin.Plugin.Tvdb/ScheduledTasks/UpdateTask.cs

```diff
@@ -0,0 +1,118 @@
+        /// Initializes a new instance of the <see cref="UpdateTask"/> class.
+        /// </summary>
+        /// <param name="tvdbClientManager">TheTvdb Client Manager.</param>
+        /// <param name="libraryManager">Library Manager.</param>
+        /// <param name="providerManager">Provider Manager.</param>
+        /// <param name="fileSystem">File System.</param>
+        public UpdateTask(TvdbClientManager tvdbClientManager, ILibraryManager libraryManager, IProviderManager providerManager, IFileSystem fileSystem)
+        {
+            _libraryManager = libraryManager;
+            _providerManager = providerManager;
+            _fileSystem = fileSystem;
+            _tvdbClientManager = tvdbClientManager;
+        }
+
+        /// <inheritdoc/>
+        public string Name => "Check for metadata updates.";
+
+        /// <inheritdoc/>
+        public string Key => "CheckForMetadataUpdatesTask";
+
+        /// <inheritdoc/>
+        public string Description => "Checks TheTvdb's API Update endpoint for updates periodically and updates metadata accordingly.";
+
+        /// <inheritdoc/>
+        public string Category => "TheTVDB";
+
+        private static int MetadataUpdateInHours => TvdbPlugin.Instance?.Configuration.MetadataUpdateInHours * -1 ?? -1;
+
+        /// <inheritdoc/>
+        public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+        {
+            progress.Report(0);
+            var toUpdateItems = await GetItemsUpdated(cancellationToken).ConfigureAwait(false);
+            progress.Report(10);
+            MetadataRefreshOptions refreshOptions = new MetadataRefreshOptions(new DirectoryService(_fileSystem))
+            {
+                MetadataRefreshMode = MetadataRefreshMode.FullRefresh
+            };
+            double increment = 90.0 / toUpdateItems.Count;
+            double currentProgress = 10;
+            foreach (BaseItem item in toUpdateItems)
+            {
+                await _providerManager.RefreshSingleItem(
+                    item,
+                    refreshOptions,
+                    cancellationToken).ConfigureAwait(false);
+                currentProgress += increment;
+                progress.Report(currentProgress);
+            }
+
+            progress.Report(100);
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<TaskTriggerInfo> GetDefaultTriggers()
+        {
+            return Enumerable.Empty<TaskTriggerInfo>();
+        }
+
+        /// <summary>
+        /// Gets all items that have been updated.
+        /// </summary>
+        /// <returns>List of items that have been updated.</returns>
+        private async Task<List<BaseItem>> GetItemsUpdated(CancellationToken cancellationToken)
+        {
+            InternalItemsQuery query = new InternalItemsQuery
+            {
+                HasTvdbId = true
+            };
+
+            List<BaseItem> itemList = _libraryManager.GetItemList(query);
```

Comment:
This will pull every single item from the database that has a tvbdb id attached.

Reply:
Hey,I did this because I am abit unsure on the functions of the library manager. Any recommendations on  how to retrieve episode/series with via the tvdbid?

Reply:
I see there is a HasAnyProviderId option in the InternalItemsQuery which I can use to search using the TheTvdb id. But i'll need to iterate through the episodeUpdates and seriesUpdates lists. Not sure if doing this is desirable?

Reply:
```
            List<BaseItem> toUpdateItems = new List<BaseItem>();

            Dictionary<string, string> providerIdPair = new Dictionary<string, string>() { { providerId, string.Empty } };
            InternalItemsQuery query = new InternalItemsQuery();

            foreach (EntityUpdate update in allUpdates)
            {
                providerIdPair[providerId] = update.RecordId!.Value.ToString(CultureInfo.InvariantCulture);
                query.HasAnyProviderId = providerIdPair;
                List<BaseItem> itemList = _libraryManager.GetItemList(query);
                if (itemList.Count > 0 && !toUpdateItems.Contains(itemList[0]))
                {
                    toUpdateItems.Add(itemList[0]);
                }
            }

```

Alternative implementation if this is better?